src/scripts/datalayer.js

```diff
@@ -371,6 +449,15 @@ governing permissions and limitations under the License.
         that._valid = !!that._type;
     };
 
+    /**
+     * Returns the index of the item in the array of existing items (added with the standard Array.prototype.push())
+     *
+     * @returns {Number} The index of the item in the array of existing items if it exists, -1 otherwise.
+     */
+    DataLayer.Item.prototype.getIndex = function() {
```

Comment:
@jckautzmann - what do we need this for? I would be tempted to remove it.

Reply:
this is needed when a listener is triggered for past events (scope = past): we loop over all past events until this listener is reached.